miden-lib/asm/miden/kernels/tx/account.masm

```diff
@@ -557,6 +557,33 @@ export.authenticate_procedure
     end
 end
 
+#! Verifies that the procedure root is part of the account code and applies the offset
+#! associated with this procedure to the provided slot index.
+#!
+#! Panics:
+#!   - if the invocation of the kernel procedure does not originate from the account context.
+#!   - slot index is out of bounds
+#!   - computed index is out of bounds
+#!
+#! Stack: [PROC_ROOT, slot_index]
+#! Output: [offset_slot_index]
+#!
+#! - PROC_ROOT is the hash of the procedure to authenticate.
+export.authenticate_storage_access
+    # verify that slot_index is in bounds, types commitment is stored at the
+    # last available location in storage i.e. 255
+    dup.4 exec.get_slot_types_commitment_storage_slot lt assert.err=111
```

Comment:
For error codes here and on line 583 below, could we use constants rather than inlined values?

Also, I'd probably change `get_slot_types_commitment_storage_slot` to something like `get_max_storage_slot_index`.

Reply:
called `Result::unwrap()` on an `Err` value: FailedAssertion { clk: 1259, err_code: 131150, err_msg: None } 

I added an custom error with a message 131150, should I receive the message in the err_msg field or did I forget to add something?

Reply:
If you've used the same IDs and added entries to the error message array it should work. One potential thing to keep in mind is that sometimes error codes are specified in hex and sometimes in decimal.